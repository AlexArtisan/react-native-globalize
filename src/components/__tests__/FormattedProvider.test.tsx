/*!
 * React Native Globalize
 *
 * Copyright 2015-2020 Josh Swan
 * Released under the MIT license
 * https://github.com/joshswan/react-native-globalize/blob/master/LICENSE
 */

import React from 'react';
import renderer from 'react-test-renderer';
import {
  FormattedCurrency,
  FormattedProvider,
  FormattedUnit,
} from '..';

describe('<FormattedProvider />', () => {
  test('renders correctly', () => {
    const tree = renderer.create((
      <FormattedProvider locale="en">
        <FormattedUnit
          unit="mile-per-hour"
          value={75}
        />
      </FormattedProvider>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('updates globalize when props change', () => {
    let root;

    renderer.act(() => {
      root = renderer.create((
        <FormattedProvider locale="en" currency="USD">
          <FormattedCurrency value={10} />
          <FormattedUnit unit="mile-per-hour" value={75} />
        </FormattedProvider>
      ));
    });

    expect(root.toJSON()).toMatchSnapshot();

    renderer.act(() => {
      root = renderer.create((
        <FormattedProvider locale="es" currency="USD">
          <FormattedCurrency value={10} />
          <FormattedUnit unit="mile-per-hour" value={75} />
        </FormattedProvider>
      ));
    });

    expect(root.toJSON()).toMatchSnapshot();

    renderer.act(() => {
      root = renderer.create((
        <FormattedProvider locale="es" currency="EUR">
          <FormattedCurrency value={10} />
          <FormattedUnit unit="mile-per-hour" value={75} />
        </FormattedProvider>
      ));
    });

    expect(root.toJSON()).toMatchSnapshot();
  });
});
