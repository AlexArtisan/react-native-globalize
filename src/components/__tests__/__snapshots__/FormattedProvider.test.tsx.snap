// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormattedProvider /> renders correctly 1`] = `
<Text
  accessibilityLabel=""
  accessible={true}
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
>
  75 miles per hour
</Text>
`;

exports[`<FormattedProvider /> updates globalize when props change 1`] = `
Array [
  <Text
    accessibilityLabel=""
    accessible={true}
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
  >
    $10.00
  </Text>,
  <Text
    accessibilityLabel=""
    accessible={true}
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
  >
    75 miles per hour
  </Text>,
]
`;

exports[`<FormattedProvider /> updates globalize when props change 2`] = `
Array [
  <Text
    accessibilityLabel=""
    accessible={true}
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
  >
    10,00 US$
  </Text>,
  <Text
    accessibilityLabel=""
    accessible={true}
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
  >
    75 millas por hora
  </Text>,
]
`;

exports[`<FormattedProvider /> updates globalize when props change 3`] = `
Array [
  <Text
    accessibilityLabel=""
    accessible={true}
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
  >
    10,00 €
  </Text>,
  <Text
    accessibilityLabel=""
    accessible={true}
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
  >
    75 millas por hora
  </Text>,
]
`;
